src/Sarif.Sarifer/BackgroundAnalysisService.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Sarifer
+{
+    /// <summary>
+    /// Performs static analysis in the background.
+    /// </summary>
+    // TODO: Reanalyze when buffer changes.
+    // TODO: Remove error list items when buffer closes.
+    [Export(typeof(IBackgroundAnalysisService))]
+    internal class BackgroundAnalysisService : IBackgroundAnalysisService
+    {
+#pragma warning disable CS0649 // Filled in by MEF
+#pragma warning disable IDE0044 // Assigned by MEF
+        [ImportMany]
+        private IEnumerable<IBackgroundAnalyzer> analyzers;
+#pragma warning restore IDE0044
+#pragma warning restore CS0649
+
+        /// <inheritdoc/>
+        public void StartAnalysis(string path, string text)
+        {
+            if (this.analyzers.Any() == true)
```

Comment:
>== true [](start = 37, length = 7)

Not needed

Reply:
Can the import be null? (Honestly don't know, what if there are none?)

---
In reply to: [519917088](https://github.com/microsoft/sarif-visualstudio-extension/pull/260#discussion_r519917088) [](ancestors = 519917088)

Reply:
I don't think it can. In my experience, whenever I've made a mistake that failed to satisfy an import, the class was never instantiated.

---
In reply to: [519917408](https://github.com/microsoft/sarif-visualstudio-extension/pull/260#discussion_r519917408) [](ancestors = 519917408,519917088)